pandas/tests/groupby/test_function.py

```diff
@@ -1,7 +1,6 @@
 import builtins
-import datetime as dt
+import datetime
```

Comment:
can you separate this change into another PR for discussion. (keep this PR as removing itertools)

The reason is that we only have 7 occurrences of `import datetime as dt` in the codebase and there is some discrepancy between which datetime of datetime.datetime is aliased as dt

The more common pattern is like `from datetime import date, datetime, time, timedelta, tzinfo`

probably best with these cleanups, is to pick one type of change at a time.

Reply:
> probably best with these cleanups, is to pick one type of change at a time.

and then after appropriate discussion can apply the changes across the complete codebase.